src/Microsoft.AspNetCore.Dispatcher/DispatcherOptions.cs

```diff
@@ -1,12 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Http;
 using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Dispatcher
 {
     public class DispatcherOptions
     {
-        public IList<DispatcherEntry> DispatcherEntryList { get; set; }
+        public IList<RequestDelegate> Dispatchers { get; } = new List<RequestDelegate>();
```

Comment:
I think that this will end up needing to be a custom collection type or a builder. Right now the delegate is the primitive.. that's all you need. **BUT** I want to have objects so we can do diagnostics and things like route debugging properly.

It needs to be possible to go bare metal, but our implementations should have rich diagnostics and you only get that by having a richer set of interfaces.